src/node/rpc/frontend.h

```diff
@@ -455,18 +455,27 @@ namespace ccf
         {
           case HandlerRegistry::Read:
           {
+            if (ctx->session.is_forwarded)
+            {
+              return forward_or_redirect_json(ctx);
+            }
             break;
           }
 
           case HandlerRegistry::Write:
           {
+            ctx->session.is_forwarded = true;
```

Comment:
Is this right? If a session connects to Primary `0` and issues a `Write` command, then if `0` becomes follower (assuming that the session stays open during the election), all subsequent `Read` on that session will be forwarded to the new Primary. It doesn't break session consistency but it doesn't seem optimal.

Reply:
I didn't see that this code was guarded by `if (!is_primary)` which makes the scenario described above impossible. However, if `0` starts as follower, then becomes primary, and then again follower, all subsequent `Read` will be forwarded.

Reply:
So, @eddyashton and I just talked about it, and we've come to the conclusion that it's fine, because what matter is _global_ read and write monotonicity. Local reads and writes can always get rolled back of course.

Reply:
I agree with that. However, it seems that forwarding all `Read`s after an election is not great in terms of scalability (i.e. all `Read`s forwarded to the primary). Should the flag be reset after an election?

Reply:
So the right strategy to reset this is not elections, it's when:

1. there are no outstanding requests (ie. requests that the node has forwarded and is waiting for an answer to)
2. the local view of the global state has caught up to the last observed global state on that session.

I suggest this is a separate PR. I also don't believe an election is sufficient to reset the flag, the node may still be behind the new leader.